.github/pull_request_template.md

```diff
@@ -0,0 +1,35 @@
+#### What does the PR do?
+_Describe your pull request here. Please read the text below the line, and make sure you follow the checklist._
+
+
+#### Checklist
+- [ ] Populated all fields in this template.
+- [ ] The PR title reflects the change.
+- [ ] Changes are described in the pull request, or an [existing issue is referenced](https://github.com/triton-inference-server/server/issues).
+- [ ] Added [test plan](#test-plan) and verified test passes.
+- [ ] Verified that the PR passes existing CI.
+- [ ] Added _succinct_ git squash message before merging [ref](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
```

Comment:
```suggestion
- [ ] Added _succinct_ git squash message before merging [ref](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
- [ ] CodeQL suggestions resolved
- [ ] Lint Gods are pleased ( i.e. pre-commit checks are green )
```

Reply:
I think this can be checked by GitHub action.

Reply:
I would remove anything from the checklist that can be verified by PR rules in Github

Reply:
My point was: in case of external contribution, we need to make contributors aware that these checks should pass. For example, multi-LoRA PR contributors ignored them until I pointed them.

Reply:
external contributors are a bit tricky - as they can't run them until we run them - so  I can understand how they could initially ignore them.... don't have a good solution here.

Reply:
Here's Llama-index example: https://github.com/run-llama/llama_index/pull/13060 
They have requirement in the template, as well as Lint checks in the github action. 
As a result, merge in first attempt, since template identified those requirements.

Reply:
agreed, ``` I ran make format; make lint to appease the lint gods```

We can update to say ``pre-commit install, pre-commit run --all`` ?

Reply:
agree